ethereum-side/NFTLock.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.6;
+
+import "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+
+import "./INFTLock.sol";
+
+contract NFTLock is INFTLock, IERC721Receiver {
+
+    struct NFT {
+        uint externalTokenId;
+        address tokenContractAddress;
+        State state;
+    }
+
+    mapping (address => mapping (uint256 => NFT)) tokens;
+    mapping (address => uint256) totalTokensByAddress;
+
+    uint256 totalDepositedTokens;
```

Comment:
Well I agree with you for our functional implementation we don't need it, but I was thinking it could be nice to show in the front how many NFTs are locked actually

Reply:
Yes, ok 👍